internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -116,6 +116,36 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 	}
 
+	// Set Taint to a node in an old machineDeployment and unset Taint from a node in a new machineDeployment
+	_, machineHasAnnotation := machine.Annotations[clusterv1.OldDeploymentAnnotation]
+	taint := corev1.Taint{Key: clusterv1.OldDeploymentAnnotation, Effect: corev1.TaintEffectPreferNoSchedule}
```

Comment:
I'd expect the taint key to be something like `ToBeDeletedByClusterAPI`

Reply:
In line with the comment above, I would suggest "cluster.x-k8s.io/outdated-revision" or "cluster.x-k8s.io/old-revision"